volkswagencarnet/vw_vehicle.py

```diff
@@ -709,32 +742,32 @@ def is_model_image_supported(self) -> bool:
     @property
     def parking_light(self) -> bool:
         """Return true if parking light is on."""
-        response = int(self.attrs.get("StoredVehicleDataResponseParsed")[P.PARKING_LIGHT].get("value", 0))
-        return response != 2
+        lights = self.attrs.get("vehicleLights").get("lightsStatus").get("value").get("lights")
```

Comment:
The `self.attrs` is a property shortcut to `self._state`, which is a dictionary.

If you drill down the dictionary using a `.get`, and the key is absent - it would return a `None`. The chained `.get` would attempt to use the `.get` on a `None`, resulting in an `AttributeError`.

You can avoid this by using a second parameter with an empty dictionary, like this: `.get(key, {})`, or opt in for square brackets, and handle the `KeyError` exception.

Reply:
Just noticed, there is a helper function `find_path`, that could be used instead.

Reply:
I noticed that one after the first few attributes and am using it consequently now. Just didn’t also refactor those first few yet. Also in the old code the usage was very mixed.
But is_valid_path and find_path make for much more readable code, for sure.